driver/annotations.md

```diff
@@ -0,0 +1,224 @@
+  the AST node starts.
+
+- ColumnKey (string): this is the key used in the native AST to indicate 
+  the column number where the code mapped to the AST node starts.
+
+- TokenKeys (map[string]bool): this map establishes what properties (as in JSON
+  keys) in the native AST nodes can be mapped to Tokens in the UAST. If the
+  InternalTypeKey is the "type" of a node, the Token could be tough of as the
+  "value" representation; this could be a specific value for string/numeric
+  literals or the symbol name for others.  E.g.: if a native AST represents a
+  numeric literal as: `{"ast_type": NumLiteral, "value": 2}` then you should have
+  to add `"value": true` to the TokenKeys map.  Some native ASTs will use several
+  different fields as tokens depending on the node type; in that case, all should
+  be added to this map to ensure a correct UAST generation.
+
+- SyntheticTokens (map[string][string]): sometimes native ASTs just use an
+  InternalTypeKey for some node but we need to add a Token to the UAST node to
+  improve the representation. In this case we can add both the InternalKey and
+  what token it should generate. E.g.: an InternalTypeKey called "NullLiteral" in
+  Java should be mapped using this map to "null" adding ```"NullLiteral":
+  "null"``` to this map.
+
+- TopLevelIsRootNode (bool): this boolean switch specifies if the parsing must be
+  start with the first node found at the top level (`true`) or one of their children
+  (`false`). This is needed because some native ASTs start with "AST" or some
+  other nonsignificant keys at the top level.
+
+- PromotedPropertyLists (map[string]map[string]bool): By default the UAST
+  generation will set as children of a node any object that hangs from any of the
+  original native AST node properties. In this process, object key serving as
+  the parent is lost and its name is added as the "internalRole" key of the children.
+  This is usually fine since the InternalTypeKey of the parent AST node will
+  usually provide enough context and the node won't any other children. This map
+  allows you to change this default behavior for specific nodes so the properties
+  are "promoted" to a new node (with an InternalTypeKey named "Parent.KeyName")
+  and the objects in its list will be shown in the UAST as children. E.g.: if you
+  have a native AST where an "If" node has the JSON keys "body", "else" and
+  "condition" each with its own list of children, you could add an entry to
+  PromotedPropertyLists like:
+
+  ```
+  "If": {"body": true, "orelse": true, "condition": true},
+  ```
+
+  In this case, the new nodes will have the InternalTypeKey "If.body", "If.orelse"
+  and "If.condition" and with these names you should be able to write specific 
+  matching rules in the annotation.go file.
+
+### UASTParserBuilder function
+
+This automatically generated function will return a driver.UASTParser object that
+will be constructed using your provided ObjectToNoder struct. This is the function
```

Comment:
You have to provide a `ToNoder`, and `ObjectToNoder` is one of the possible implementations you could use.